### PR TITLE
fix(schemas): payload template definition was too restrictive

### LINKED ISSUE #101: Validation error in Parameters for Task type Pass
**Describe the bug**
For State Type Pass it throws validation errror
```
 keyword: 'type',
    dataPath: ".States['EmptyState'].Parameters[0]",
    schemaPath: 'pass.json#/properties/Parameters/items/type',
    params: { type: 'object' },
    message: 'should be object'
```
**To Reproduce**
Please provide a complete state machine definition:
```json
{
  "StartAt": "EmptyState",
  "States": {
    "EmptyState": {
       "Type": "Pass",
      "Parameters": ["abc"],
      "ResultPath": "$.emptyState",
      "End": true
    }
  }
}
```

**Expected behavior**

Parameters can accept ARRAY, NUMBER, BOOLEAN, OBJECT, STRING

**Version:** [e.g. 1.9.4, can be found by running `asl-validator --version`]
2.2.1
**Additional context**

### PR DESCRIPTION
**Pull request by massfords**
closes #101

### CODE REVIEW

File: src/__tests__/error-reporting.test.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ describe("tests with definitions containing errors to see what's reported", () =
                     schemaError: {
                         instancePath: "/States/Hello, World/Parameters/lorem.$",
                         // the regular expression in the path is escaped by AJV when it reports the location of the error
-                        schemaPath: "#/oneOf/0/patternProperties/%5E.%2B%5C.%5C%24%24/format"
+                        schemaPath: "#/patternProperties/%5E.%2B%5C.%5C%24%24/format"
```
Comment (massfords) [inline src/__tests__/error-reporting.test.ts:47]: incidental change due to the schema refactoring

File: src/schemas/paths.json [hunk 2]
```diff
@@ -10,48 +10,67 @@
...
     },
+    "_payload_template_object": {
+      "type": "object",
+      "patternProperties": {
+        "^.+\\.\\$$": {
+          "$comment": "matches fields ending in .$",
+          "type": "string",
+          "nullable": true,
+          "format": "asl_payload_template"
+        },
```
Comment (massfords) [inline src/schemas/paths.json:21]: extracted this to its own definition to better reuse below.

File: src/schemas/paths.json [hunk 3]
```diff
@@ -10,48 +10,67 @@
...
         {
           "type": "array",
           "items": {
             "$ref": "#/definitions/asl_payload_template"
           }
+        },
+        {
+          "type": ["string", "boolean", "number"]
+        }
+      ]
```
Comment (massfords) [inline src/schemas/paths.json:60]: allowing for the additional simple types here

**Review by ChristopheBougere — APPROVED** [review]
Looks good to me 👍

**Pull request merged**